### ISSUE TITLE
Testing epic saved carts

### ISSUE THREAD

**Issue by dydome** [SAP/spartacus#11518]
Environment configuration:

- url: https://spartacus-dev3.eastus.cloudapp.azure.com:9002
- b2b: true

Branch: epic/saved-carts

Scenarios:

- [x] Saving cart for later as logged user
- [x] ROUND 2 after refactoring
1. Add any product to the cart
2. Open cart page
3. Click "Save cart for later"
4. Dialog form should appear with empty fields
5. After success save cart should refresh and the global message should appear

- [x] Navigate to saved cart list from cart
- [x] ROUND 2 after refactoring
1. Add any product to the cart
2. Open cart page
3. Click "Saved carts"
4. Should navigate to saved carts

- [x] Navigate to saved cart list from account menu  as logged user
- [x] ROUND 2 after refactoring
1. Click on the My Account dropdown
2. Click "Saved carts"
3. Should navigate to saved carts

- [x] Navigate to login page if a user is not logged in after clicking "save cart for later" or "saved carts" on the cart page

- [x] Restore saved cart from saved carts list
- [x] ROUND 2 after refactoring
1. Cart should be empty
2. Go to saved carts list
3. Click "make cart active"
4. List should be refreshed and a cart filled with new entries

- [x] Restore saved cart with active cart from saved carts list
- [x] ROUND 2 after refactoring
1. Cart should contain some entries
2. Go to saved carts list
3. Click "make cart active"
4. List should be refreshed and a cart filled with new entries
5. Previously active cart should be saved and appear on the list with name as ID

- [x] Open saved cart details
- [x] ROUND 2 after refactoring
1. Go to saved carts list
2. Click on any element in the row to navigate to cart details

- [x] Restore saved cart from saved carts details
- [x] ROUND 2 after refactoring
1. Cart should be empty
2. Go to saved carts details
3. Click "make cart active"
4. Cart should be filled with new entries
5. Should navigate to saved carts list

- [x] Restore saved cart from saved carts details with active cart from saved carts list
- [x] ROUND 2 after refactoring
1. Cart should contain some entries
2. Go to saved carts details
3. Click "make cart active"
4. Cart should be filled with new entries
5. Should navigate to saved carts list

- [x] Edition of the saved cart
- [x] ROUND 2 after refactoring
1. Go to saved carts details
2. Click on the "pencil" icon button next to the cart name
3. Dialog form should appear with filled values from details
4. After clicking save, dialog should close and details on the page should be refreshed

- [x] Saving previously restored cart
- [x] ROUND 2 after refactoring
1. Save new cart for later with name and description
2. Go to saved carts list
3. Click "make cart active"
4. Go to the cart page
5. Click "Save cart for later"
6. Dialog form should appear with filled values from details

- [x] Delete saved cart
- [x] ROUND 2 after refactoring
1. Go to saved carts details
2. Click "delete saved cart"
3. Dialog form should appear with filled values from details
4. Click 'Delete"
5. After clicking delete, dialog should close and should navigate to saved carts list

- [x] Can perform checkout with a restored cart or a swapped cart 
- [x] ROUND 2 after refactoring
1. do a checkout with a restored cart 
2. do a checkout when you swap and swap back to the cart
3. do a checkout for replenishment
4. do a checkout with any other config with the restored/swapped cart

**Comment by mirkosoft** [comment]
There is no link to "saved carts" when actual cart is empty: bug or feature? I can still open that page though:
http://localhost:4200/powertools-spa/en/USD/my-account/saved-carts (there's also a link in the account menu ofc)

**Comment by mirkosoft** [comment]
After removing item from an active cart (in order to have it empty) and activating another saved cart, the previous cart (which is empty now) is still on the list of saved carts and can be accessed (while it has no entries):
![image](https://user-images.githubusercontent.com/788092/111625841-9b73f980-87ed-11eb-8da6-23ab1532457c.png)
It's also possible to activate this empty cart as well.

**Comment by mirkosoft** [comment]
Quantity is wrong on the saved carts page (it's cart's `totalItems` on the list but maybe we should have there quantity as well, ie: "Qty/Total items" as column title and "1/3" in the row value; or just add the another column):
![image](https://user-images.githubusercontent.com/788092/111626718-a1b6a580-87ee-11eb-8420-a84dac68e33c.png)
![image](https://user-images.githubusercontent.com/788092/111626812-b5620c00-87ee-11eb-822d-0d71d42de4ed.png)

**Comment by dydome** [comment]
> After removing item from an active cart (in order to have it empty) and activating another saved cart, the previous cart (which is empty now) is still on the list of saved carts and can be accessed (while it has no entries):
> ![image](https://user-images.githubusercontent.com/788092/111625841-9b73f980-87ed-11eb-8da6-23ab1532457c.png)
> It's also possible to activate this empty cart as well.

Will be fixed in this PR -> https://github.com/SAP/spartacus/pull/11542

**Comment by dydome** [comment]
Wrong value on the list for quantity column will be fixed in -> https://github.com/SAP/spartacus/pull/11561

**Comment by bgambocjaviniar** [comment]
> There is no link to "saved carts" when actual cart is empty: bug or feature? I can still open that page though:
> http://localhost:4200/powertools-spa/en/USD/my-account/saved-carts (there's also a link in the account menu ofc)

feature

**Comment by bgambocjaviniar** [comment]
Keeping this open for round 2 testing after. Will be INCLUDING a second checkox for it right beside the first one.
Added the test case of checking out with the saved cart

**Comment by bgambocjaviniar** [comment]
> Wrong value on the list for quantity column will be fixed in -> #11561

wasn't fixed in the pr and it got merged.
Including it here instead https://github.com/SAP/spartacus/pull/11597

accelerator uses `qty/quantity` in list and details page as `totalItems`. It doesn't have the property `items`, but in Spartacus, we included it in the details page. I made it so that `qty/quantity` is set as totalItems as done in accelerator and for items as `totalUnitCount`.
